A fun new error with `-Werror` in Python 3.11.3 that I cannot reproduce locally:
```
FAILED: numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_override.c.o 
cc -Inumpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p -Inumpy/core -I../numpy/core -Inumpy/core/include -I../numpy/core/include -I../numpy/core/src/common -I../numpy/core/src/multiarray -I../numpy/core/src/npymath -I../numpy/core/src/umath -I/usr/include/x86_64-linux-gnu/openblas-serial/ -I/opt/hostedtoolcache/Python/3.11.3/x64/include/python3.11 -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -Werror -std=c99 -O2 -g -fno-strict-aliasing -DHAVE_CBLAS -fPIC -DNPY_INTERNAL_BUILD -DHAVE_NPY_CONFIG_H -DNPY_DISABLE_OPTIMIZATION -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -MD -MQ numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_override.c.o -MF numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_override.c.o.d -o numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so.p/src_umath_override.c.o -c ../numpy/core/src/umath/override.c
In file included from /opt/hostedtoolcache/Python/3.11.3/x64/include/python3.11/Python.h:44,
                 from ../numpy/core/include/numpy/npy_3kcompat.h:15,
                 from ../numpy/core/src/common/npy_pycompat.h:4,
                 from ../numpy/core/src/umath/override.c:4:
../numpy/core/src/umath/override.c: In function ‘PyUFunc_CheckOverride’:
/opt/hostedtoolcache/Python/3.11.3/x64/include/python3.11/object.h:538:9: error: ‘override_array_ufunc’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  538 |         _Py_Dealloc(op);
      |         ^~~~~~~~~~~~~~~
../numpy/core/src/umath/override.c:319:19: note: ‘override_array_ufunc’ was declared here
  319 |         PyObject *override_array_ufunc;
      |                   ^~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/numpy/core/src/umath/override.c
+++ b/numpy/core/src/umath/override.c
@@ -315,10 +315,9 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
 
     /* Call __array_ufunc__ functions in correct order */
     while (1) {
-        PyObject *override_obj;
-        PyObject *override_array_ufunc;
+        PyObject *override_obj = NULL;
+        PyObject *override_array_ufunc = NULL;
 
-        override_obj = NULL;
         *result = NULL;
 
         /* Choose an overriding argument */

